pkg/vault/vaultcontainer.go

```diff
@@ -106,10 +106,24 @@ func valideSecretPaths(paths []nais.SecretPath) error {
 
 func (c config) addVaultContainer(spec *corev1.PodSpec, paths []nais.SecretPath) (*corev1.PodSpec, error) {
 
-	if err := valideSecretPaths(paths); err != nil {
+	if err := validateSecretPaths(paths); err != nil {
 		return nil, err
 	}
 
+	if c.app.Spec.Vault.LoadDefault {
+		var defaultPathExists = false
+		for _, path := range paths {
+			defaultMountPathExists := filepath.Clean(nais.DefaultVaultMountPath) == filepath.Clean(path.MountPath)
+			defaultKvPathExists := filepath.Clean(c.defaultSecretPath().KvPath) == filepath.Clean(path.KvPath)
+			if defaultMountPathExists || defaultKvPathExists {
+				defaultPathExists = true
```

Comment:
Remove defaultPathExist, inverse if check and append default path if check is true is easier to.read imo

Reply:
Don't we have to iterate through all the paths for the check? Otherwise we'd just be appending the default path for every path that doesn't match